spot_driver/test/test_spot_driver.py

```diff
@@ -126,6 +136,489 @@ def test_wrapped_commands(self) -> None:
+
+        feedback.feedback.arm_drag_feedback.status.value = feedback.feedback.arm_drag_feedback.status.STATUS_DRAGGING
+        self.assertEqual(self.spot_ros2._process_synchronized_arm_command_feedback(feedback), GoalResponse.SUCCESS)
+
+        feedback.feedback.arm_drag_feedback.status.value = feedback.feedback.arm_drag_feedback.status.STATUS_UNKNOWN
+        self.assertEqual(self.spot_ros2._process_synchronized_arm_command_feedback(feedback), GoalResponse.FAILED)
+
+        feedback.feedback.arm_drag_feedback.status.value = (
+            feedback.feedback.arm_drag_feedback.status.STATUS_GRASP_FAILED
+        )
+        self.assertEqual(self.spot_ros2._process_synchronized_arm_command_feedback(feedback), GoalResponse.FAILED)
+
+        feedback.feedback.arm_drag_feedback.status.value = (
+            feedback.feedback.arm_drag_feedback.status.STATUS_OTHER_FAILURE
+        )
+        self.assertEqual(self.spot_ros2._process_synchronized_arm_command_feedback(feedback), GoalResponse.FAILED)
+
+        """ Testing arm drag feedback """
+        feedback.feedback.feedback_choice = feedback.feedback.FEEDBACK_ARM_IMPEDANCE_FEEDBACK_SET
+
+        self.assertEqual(self.spot_ros2._process_synchronized_arm_command_feedback(feedback), None)
+
+        """ Testing arm Impedance feedback """
+        feedback.feedback.feedback_choice = feedback.feedback.FEEDBACK_ARM_IMPEDANCE_FEEDBACK_SET
+        self.assertEqual(self.spot_ros2._process_synchronized_arm_command_feedback(feedback), None)
+
+        feedback.feedback.feedback_choice = feedback.feedback.FEEDBACK_NOT_SET
+
+        self.assertEqual(self.spot_ros2._process_synchronized_arm_command_feedback(feedback), None)
+
+        """ Testing unknown arm command """
+        feedback.feedback.feedback_choice = FEEDBACK_INVALID
+
+        self.assertEqual(self.spot_ros2._process_synchronized_arm_command_feedback(feedback), GoalResponse.IN_PROGRESS)
+
+    def test_process_synchronized_mobility_command_feedback(self) -> None:
+        """Testing mobility commands"""
+        feedback = MobilityCommandFeedback()
+
+        feedback.status.value = feedback.status.STATUS_COMMAND_OVERRIDDEN
+        self.assertEqual(self.spot_ros2._process_synchronized_mobility_command_feedback(feedback), GoalResponse.FAILED)
+
+        feedback.status.value = feedback.status.STATUS_COMMAND_TIMED_OUT
+        self.assertEqual(self.spot_ros2._process_synchronized_mobility_command_feedback(feedback), GoalResponse.FAILED)
+
+        feedback.status.value = feedback.status.STATUS_ROBOT_FROZEN
+        self.assertEqual(self.spot_ros2._process_synchronized_mobility_command_feedback(feedback), GoalResponse.FAILED)
+
+        feedback.status.value = feedback.status.STATUS_INCOMPATIBLE_HARDWARE
+        self.assertEqual(self.spot_ros2._process_synchronized_mobility_command_feedback(feedback), GoalResponse.FAILED)
+
+        feedback.status.value = feedback.status.STATUS_PROCESSING
```

Comment:
Should there be a test under this?

Reply:
You have to set the status to `STATUS_PROCESSING` before evaluating commands so we don't need a test to directly follow setting this status value.